python/mxnet/visualization.py

```diff
@@ -271,14 +275,21 @@ def plot_network(symbol, title="plot", save_format='pdf', shape=None, node_attrs
         raise ImportError("Draw network requires graphviz library")
     if not isinstance(symbol, Symbol):
         raise TypeError("symbol must be a Symbol")
+    internals = symbol.get_internals()
     draw_shape = False
     if shape is not None:
         draw_shape = True
-        interals = symbol.get_internals()
-        _, out_shapes, _ = interals.infer_shape(**shape)
+        _, out_shapes, _ = internals.infer_shape(**shape)
         if out_shapes is None:
             raise ValueError("Input shape is incomplete")
-        shape_dict = dict(zip(interals.list_outputs(), out_shapes))
+        shape_dict = dict(zip(internals.list_outputs(), out_shapes))
+    draw_type = False
+    if dtype is not None:
+        draw_type = True
+        _, out_types, _ = internals.infer_type(**dtype)
+        if out_types is None:
+            raise ValueError("Input type is incomplete")
```

Comment:
maybe offer a suggestion on what more types to add. related info can be found from list_inputs()

Reply:
It is more complicated though as you don't really need to provide type for all the inputs. The actual solution for this problem would be to implement infer_type_partial (similar to infer_shape_partial) and use it here to understand which shapes and types could not be inferred.
I think this would be good to do in another PR.

Reply:
OK, that's fine